status/node_controller.go

```diff
@@ -0,0 +1,89 @@
+package status
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/awslabs/operatorpkg/object"
+	"github.com/samber/lo"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type node struct {
+	*corev1.Node
+}
+
+func (n node) GetConditions() []Condition {
+	return lo.Map(n.Status.Conditions, func(condition corev1.NodeCondition, _ int) Condition {
```

Comment:
Rather than directly knowing what type of object this is, can we use unstructured to make this work generally for every type of objct?